BlueprintUI/Sources/Layout/Layout.swift

```diff
@@ -33,10 +129,172 @@ public protocol Layout {
+    ///       ``SizeConstraint/Axis/unconstrained``. Otherwise, your behavior would be in violation
+    ///       of rule #1.
+    ///
+    /// - Parameters:
+    ///   - proposal: A size constraint for the container. The container's parent element that calls
+    ///     this method might call the method more than once with different constraints to learn
+    ///     more about the container’s flexibility before choosing a size for placement.
+    ///   - subelements: A collection of proxies that represent the elements that the container
+    ///     arranges. You can use the proxies in the collection to get information about the
+    ///     subelements as you determine how much space the container needs to display them.
+    ///   - cache: Optional storage for calculated data that you can share among the methods of your
+    ///     custom layout container. See ``makeCache(subelements:)-4kmy2`` for details.
+    /// - Returns: A size that indicates how much space the container needs to arrange its
+    ///   subelements.
+    func sizeThatFits(
+        proposal: SizeConstraint,
+        subelements: Subelements,
+        cache: inout Cache
+    ) -> CGSize
+
+    /// Assigns positions to each of the layout’s subelements.
+    ///
+    /// Blueprint calls your implementation of this method to tell your custom layout container to
+    /// place its subelements. From this method, call the
+    /// ``LayoutSubelement/place(at:anchor:size:)`` method on each item in `subelements` to tell the
+    /// subelements where to appear in the user interface.
+    ///
+    /// You can also update the ``LayoutSubelement/attributes-swift.property`` property of each
+    /// subelement to set properties like opacity and transforms on each subelement.
+    ///
+    /// Be sure that you use computations during placement that are consistent with those in your
+    /// implementation of other protocol methods for a given set of inputs. For example, if you add
+    /// spacing during placement, make sure your implementation of
+    /// ``sizeThatFits(proposal:subelements:cache:)`` accounts for the extra space.
+    ///
+    /// - Parameters:
+    ///   - size: The region that the container's parent allocates to the container. Place all the
+    ///     container's subelements within the region. The size of this region may not match any
+    ///     size that was returned from a call to ``sizeThatFits(proposal:subelements:cache:)``.
+    ///   - subelements: A collection of proxies that represent the elements that the container
+    ///     arranges. Use the proxies in the collection to get information about the subelements and
+    ///     to tell the subelements where to appear.
+    ///   - cache: Optional storage for calculated data that you can share among the methods of your
+    ///     custom layout container. See ``makeCache(subelements:)-4kmy2`` for details.
+    func placeSubelements(
+        in size: CGSize,
+        subelements: Subelements,
```

Comment:
Note for readers: Later on, the writable properties here are wrapped in a `@Storage` property so you can write to them.